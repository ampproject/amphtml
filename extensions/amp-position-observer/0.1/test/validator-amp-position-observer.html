<!--
  Copyright 2017 The AMP HTML Authors. All Rights Reserved.

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS-IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the license.
-->
<!--
  Test Description:
  Tests amp-position-observer.
-->
<!doctype html>
<html âš¡>
<head>
  <meta charset="utf-8">
  <link rel="canonical" href="./regular-html-version.html">
  <meta name="viewport" content="width=device-width,minimum-scale=1">
  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-position-observer" src="https://cdn.ampproject.org/v0/amp-position-observer-0.1.js"></script>
</head>
<body>
  <!-- Valid -->
  <amp-position-observer layout="nodisplay"></amp-position-observer>
  <amp-position-observer layout="nodisplay" on="enter:foo;exit:bar;scroll:baz(percent=event.percent)"></amp-position-observer>
  <amp-position-observer layout="nodisplay" on="enter:foo;exit:bar;scroll:baz(percent=event.percent)" once></amp-position-observer>
  <amp-position-observer target="#id" layout="nodisplay"></amp-position-observer>
  <amp-position-observer target="#id" layout="nodisplay" once></amp-position-observer>
  <amp-position-observer intersection-ratios="0" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="1" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.5" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.5 0" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.5 0" layout="nodisplay" once></amp-position-observer>
  <amp-position-observer intersection-ratios="1 0.5" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100" layout="nodisplay" once></amp-position-observer>
  <amp-position-observer viewport-margins="100 200" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100vh" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100px" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100px 200" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100px 200vh" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100vh 200px" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100 200vh" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100 200vh" layout="nodisplay" once></amp-position-observer>
  <!-- Invalid: missing required layout value -->
  <amp-position-observer></amp-position-observer>
  <!-- Invalid: interesection-ratios not valid entries -->
  <amp-position-observer intersection-ratios="" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="1.1" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.b" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="foo" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="3" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.a 0.b" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.5 foo" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.5 1.1" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="0.5 4" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="foo 0.5" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="1.1 0.5" layout="nodisplay"></amp-position-observer>
  <amp-position-observer intersection-ratios="4 0.5" layout="nodisplay"></amp-position-observer>
  <!-- Invalid: viewport-margins not valid entries -->
  <amp-position-observer viewport-margins="" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="foo" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100fx" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100vx" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100 100vx" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100vh 100vx" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100fx 100vh" layout="nodisplay"></amp-position-observer>
  <amp-position-observer viewport-margins="100vx 100px" layout="nodisplay"></amp-position-observer>
</body>
</html>
