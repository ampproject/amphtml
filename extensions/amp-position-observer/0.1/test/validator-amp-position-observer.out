FAIL
|  <!--
|    Copyright 2017 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    Tests amp-position-observer.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|    <meta charset="utf-8">
|    <link rel="canonical" href="./regular-html-version.html">
|    <meta name="viewport" content="width=device-width,minimum-scale=1">
|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|    <script async src="https://cdn.ampproject.org/v0.js"></script>
|    <script async custom-element="amp-position-observer" src="https://cdn.ampproject.org/v0/amp-position-observer-0.1.js"></script>
|  </head>
|  <body>
|    <!-- Valid -->
|    <amp-position-observer layout="nodisplay"></amp-position-observer>
|    <amp-position-observer layout="nodisplay" on="enter:foo;exit:bar;scroll:baz(percent=event.percent)"></amp-position-observer>
|    <amp-position-observer layout="nodisplay" on="enter:foo;exit:bar;scroll:baz(percent=event.percent)" once></amp-position-observer>
|    <amp-position-observer target="#id" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer target="#id" layout="nodisplay" once></amp-position-observer>
|    <amp-position-observer intersection-ratios="0" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer intersection-ratios="1" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer intersection-ratios="0.5" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer intersection-ratios="0.5 0" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer intersection-ratios="0.5 0" layout="nodisplay" once></amp-position-observer>
|    <amp-position-observer intersection-ratios="1 0.5" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100" layout="nodisplay" once></amp-position-observer>
|    <amp-position-observer viewport-margins="100 200" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100vh" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100px" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100px 200" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100px 200vh" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100vh 200px" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100 200vh" layout="nodisplay"></amp-position-observer>
|    <amp-position-observer viewport-margins="100 200vh" layout="nodisplay" once></amp-position-observer>
|    <!-- Invalid: missing required layout value -->
|    <amp-position-observer></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:54:2 The implied layout 'CONTAINER' is not supported by tag 'amp-position-observer'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [AMP_LAYOUT_PROBLEM]
|    <!-- Invalid: interesection-ratios not valid entries -->
|    <amp-position-observer intersection-ratios="" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:56:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value ''. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="1.1" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:57:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '1.1'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="0.b" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:58:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '0.b'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="foo" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:59:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value 'foo'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="3" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:60:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '3'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="0.a 0.b" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:61:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '0.a 0.b'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="0.5 foo" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:62:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '0.5 foo'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="0.5 1.1" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:63:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '0.5 1.1'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="0.5 4" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:64:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '0.5 4'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="foo 0.5" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:65:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value 'foo 0.5'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="1.1 0.5" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:66:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '1.1 0.5'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer intersection-ratios="4 0.5" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:67:2 The attribute 'intersection-ratios' in tag 'amp-position-observer' is set to the invalid value '4 0.5'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <!-- Invalid: viewport-margins not valid entries -->
|    <amp-position-observer viewport-margins="" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:69:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value ''. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer viewport-margins="foo" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:70:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value 'foo'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer viewport-margins="100fx" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:71:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value '100fx'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer viewport-margins="100vx" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:72:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value '100vx'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer viewport-margins="100 100vx" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:73:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value '100 100vx'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer viewport-margins="100vh 100vx" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:74:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value '100vh 100vx'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer viewport-margins="100fx 100vh" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:75:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value '100fx 100vh'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|    <amp-position-observer viewport-margins="100vx 100px" layout="nodisplay"></amp-position-observer>
>>   ^~~~~~~~~
amp-position-observer/0.1/test/validator-amp-position-observer.html:76:2 The attribute 'viewport-margins' in tag 'amp-position-observer' is set to the invalid value '100vx 100px'. (see https://www.ampproject.org/docs/reference/components/amp-position-observer) [DISALLOWED_HTML]
|  </body>
|  </html>